### PR TITLE
Create entity from list members page

### PR DESCRIPTION
**Pull request by robjlucas**
If no entity is found when searching for an entity to add from the list members page, offers a button which opens an entity creation form in a modal.

This modal creates the entity using the new `lists/entities_controller.rb`, adds it to the list, and redirects back to the list members page.

### CODE REVIEW

File: spec/features/js/creating_entities_from_list_members_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    before do
+      setup_sphinx do
+        [oedipa, mucho].each do |person|
+          ListEntity.create!(list_id: list.id, entity_id: person.id)
+        end
+      end
+    end
+
+    after do
+      teardown_sphinx { delete_entity_tables }
```
Comment (aepyornis) [inline spec/features/js/creating_entities_from_list_members_spec.rb:28]: try running without the "delete_entity_tables"  and see if it still works?

File: spec/controllers/lists/entities_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
+describe Lists::EntitiesController, type: :controller do
```
Comment (aepyornis) [inline spec/controllers/lists/entities_controller_spec.rb:1]: Great tests! Once we have more confidence that js tests run reliably-- we could consider testing most of this with the js/feature tests

File: app/controllers/lists/entities_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+module Lists
+  class EntitiesController < ApplicationController
+    before_action :authenticate_user!
+    before_action :block_restricted_user_access
+    before_action -> { current_user.raise_unless_can_edit! }
+    before_action :set_list, :set_entity
+
+    def create
+      if @entity.save
+        # Ought to be able to do this stuff with accepts_nested_attributes_for
```
Comment (aepyornis) [inline app/controllers/lists/entities_controller.rb:12]: Agreed! Plenty of refactoring we could work on in the list and entity controllers too.

File: spec/features/js/creating_entities_from_list_members_spec.rb (hunk 1 above)
Comment (robjlucas) [reply spec/features/js/creating_entities_from_list_members_spec.rb]: It works! Thanks.

**Pull request merged**